src/analysis/complex/open_mapping.lean

```diff
@@ -0,0 +1,98 @@
+/-
+Copyright (c) 2022 Vincent Beffara. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Vincent Beffara
+-/
+import analysis.analytic.isolated_zeros
+import analysis.complex.cauchy_integral
+import analysis.complex.abs_max
+import topology.algebra.field
+import topology.locally_constant.basic
+
+/-!
+# The open mapping theorem for holomorphic functions
+
+A holomorphic function on a preconnected open set of the complex plane is either constant or open.
+-/
+
+open set filter metric complex
+open_locale topological_space
+
+variables {U : set ℂ} {f : ℂ → ℂ} {z₀ w : ℂ} {ε r m : ℝ}
+
+lemma diff_cont_on_cl.ball_subset_image_closed_ball (h : diff_cont_on_cl ℂ f (ball z₀ r))
+  (hr : 0 < r) (hf : ∀ z ∈ sphere z₀ r, ε ≤ ∥f z - f z₀∥) (hz₀ : ¬ ∀ᶠ z in 𝓝 z₀, f z = f z₀) :
+  ball (f z₀) (ε / 2) ⊆ f '' closed_ball z₀ r :=
+begin
+  rintro v hv,
+  have h2 : diff_cont_on_cl ℂ (λ z, f z - v) (ball z₀ r) := h.sub_const v,
+  have h3 : continuous_on (λ z, ∥f z - v∥) (closed_ball z₀ r),
+    from continuous_norm.comp_continuous_on (closure_ball z₀ hr.ne.symm ▸ h2.continuous_on),
+  have h4 : analytic_on ℂ f (ball z₀ r) := h.differentiable_on.analytic_on is_open_ball,
+  have h5 : ∀ z ∈ sphere z₀ r, ε / 2 ≤ ∥f z - v∥,
+    { rintro z hz,
+      have := norm_sub_sub_norm_sub_le_norm_sub (f z) v (f z₀),
+      linarith [hf z hz, (show ∥v - f z₀∥ < ε / 2, from mem_ball.mp hv)] },
+  obtain ⟨w, hw, hfw⟩ : ∃ z ∈ ball z₀ r, ∥f z - v∥ < ε / 2,
+    from ⟨z₀, mem_ball_self hr, by simpa [← dist_eq_norm, dist_comm] using mem_ball.mp hv⟩,
+  obtain ⟨z, hz1, hz2⟩ : ∃ z ∈ ball z₀ r, is_local_min (λ z, ∥f z - v∥) z,
+    from exists_local_min_mem_ball h3 h5 hw hfw,
+  have h7 := h2.differentiable_on.eventually_differentiable_at (is_open_ball.mem_nhds hz1),
+  refine ⟨z, ball_subset_closed_ball hz1, sub_eq_zero.mp _⟩,
+  refine (eventually_eq_or_eq_zero_of_is_local_min_norm h7 hz2).resolve_left (λ key, _),
+  have h8 : ∀ᶠ w in 𝓝 z, f w = f z := by { filter_upwards [key] with h; field_simp },
+  have h9 : is_preconnected (ball z₀ r) := (convex_ball z₀ r).is_preconnected,
+  have h10 : ∃ᶠ w in 𝓝[≠] z, f w = f z := (h8.filter_mono nhds_within_le_nhds).frequently,
+  have h11 := h4.eq_on_of_preconnected_of_frequently_eq analytic_on_const h9 hz1 h10,
+  have h12 : f z = f z₀ := (h11 (mem_ball_self hr)).symm,
+  exact hz₀ (mem_of_superset (ball_mem_nhds z₀ hr) (h12 ▸ h11))
+end
+
+lemma diff_cont_on_cl.continuous_on_closed_ball (hf : diff_cont_on_cl ℂ f (ball z₀ r)) :
+  continuous_on f (closed_ball z₀ r) :=
+if h : r = 0 then by simp only [h, closed_ball_zero, continuous_on_singleton]
+  else closure_ball z₀ h ▸ hf.continuous_on
```

Comment:
Isn't it already there as [diff_cont_on_cl.continuous_on_ball](https://leanprover-community.github.io/mathlib_docs/analysis/calculus/diff_on_int_cont.html#diff_cont_on_cl.continuous_on_ball)?

Reply:
(But your name is better)

Reply:
Ah yes it's the same with isomorphic proof. I missed it when I wrote my proof, either because the name was not evocative, or because it was not there yet, or because I can't search mathlib well enough (but I did look for it!)

I will use the mathlib version then.